### ISSUE TITLE
Upgrading from 1.7.2 to 1.8.0 results in IsADirectoryErrors

### ISSUE THREAD

**Issue by timthelion** [django-crispy-forms/django-crispy-forms#913]
* Package version: 1.8.0
* Django version: 2.2.6
* Python version: 3.6.8
* Template pack: (Optional)

### Description:

After the upgrade all tests fail with:

````
======================================================================
ERROR: test_dpnk_t_shirt_size_no_sizes (t_shirt_delivery.test.test_views.ViewsTestsLogon)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app-v/t_shirt_delivery/test/test_views.py", line 88, in test_dpnk_t_shirt_size_no_sizes
    self.assertRedirects(response, reverse("typ_platby"), target_status_code=200)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/testcases.py", line 371, in assertRedirects
    redirect_response = response.client.get(path, QueryDict(query), secure=(scheme == 'https'))
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/client.py", line 535, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/client.py", line 347, in get
    **extra,
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/client.py", line 422, in generic
    return self.request(**r)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/client.py", line 503, in request
    raise exc_value
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/core/handlers/base.py", line 145, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/core/handlers/base.py", line 143, in _get_response
    response = response.render()
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/response.py", line 106, in render
    self.content = self.rendered_content
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/response.py", line 83, in rendered_content
    content = template.render(context, self._request)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 987, in render
    output = self.filter_expression.resolve(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 671, in resolve
    obj = self.var.resolve(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 796, in resolve
    value = self._resolve_lookup(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 858, in _resolve_lookup
    current = current()
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 76, in super
    return mark_safe(self.render(self.context))
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 987, in render
    output = self.filter_expression.resolve(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 671, in resolve
    obj = self.var.resolve(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 796, in resolve
    value = self._resolve_lookup(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 858, in _resolve_lookup
    current = current()
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 76, in super
    return mark_safe(self.render(self.context))
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/defaulttags.py", line 309, in render
    return nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/defaulttags.py", line 309, in render
    return nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/crispy_forms/templatetags/crispy_forms_tags.py", line 208, in render
    return template.render(c)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/crispy_forms/templatetags/crispy_forms_utils.py", line 34, in render
    return remove_spaces(self.nodelist.render(context).strip())
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 188, in render
    return template.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 173, in render
    return self._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/defaulttags.py", line 309, in render
    return nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 188, in render
    return template.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 173, in render
    return self._render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/crispy_forms/templatetags/crispy_forms_utils.py", line 34, in render
    return remove_spaces(self.nodelist.render(context).strip())
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/defaulttags.py", line 209, in render
    nodelist.append(node.render_annotated(context))
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loader_tags.py", line 176, in render
    template = context.template.engine.get_template(template_name)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/engine.py", line 143, in get_template
    template, origin = self.find_template(template_name)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/engine.py", line 125, in find_template
    template = loader.get_template(name, skip=skip)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loaders/base.py", line 24, in get_template
    contents = self.get_contents(origin)
  File "/home/test/.local/share/virtualenvs/app-v-QaxYZVWa/lib/python3.6/site-packages/django/template/loaders/filesystem.py", line 23, in get_contents
    with open(origin.name, encoding=self.engine.file_charset) as fp:
IsADirectoryError: [Errno 21] Is a directory: '/app-v/dpnk/templates'
````

### Preferably also include:

I noticed in the output of my test suit the following lines:

````
EWARNING Undefined variable or unknown value for: 'field_template'
ERROR Internal Server Error: /typ_platby/
````

I hypothesize that `field_template` is getting resolved to an empty string `''` and this is causing django to try to load the templates directory as a template.

**Comment by timthelion** [comment]
Bisect says:

````
5bfc957b966aed60d6650d622dde617a717293e8 is the first bad commit
commit 5bfc957b966aed60d6650d622dde617a717293e8
Author: Rouven Bauer <r_bauer11@cs.uni-kl.de>
Date:   Thu Oct 4 12:03:06 2018 +0200

    Allow field_template to be changed with template packs (#836)
    
    * Allow field_template to be changed with template packs

:040000 040000 f9c4c7d919ab3bbd9bb0f8b4c9713a7dac429845 d28d8bd9defc9e54b1596ae23ba7084ef552754d M	crispy_forms
bisect run success
````

**Comment by timthelion** [comment]
By the way [DPNK](https://github.com/auto-mat/do-prace-na-kole) is open source and you can reproduce this with docker compose.

**Comment by carltongibson** [comment]
Thanks for that. I guess we need to see your FormHelper, and the value of `template_name` at the point the exception is raised. 

> By the way DPNK is open source and you can reproduce this with docker compose.

Somewhat time limited right now. Can you reduce it to the simplest example?

**Comment by timthelion** [comment]
I think we don't use any custom `FormHelper`s.

````
git --no-pager grep -n -e FormHelper -- \* .\[\!.\]\* ..\?\*
dpnk/auth.py:20:from crispy_forms.helper import FormHelper
dpnk/auth.py:110:        self.helper = FormHelper()
dpnk/forms.py:28:from crispy_forms.helper import FormHelper
dpnk/forms.py:102:            self.helper = FormHelper()
dpnk/forms.py:116:        self.helper = FormHelper()
dpnk/forms.py:188:        self.helper = FormHelper()
dpnk/forms.py:230:        self.helper = FormHelper()
dpnk/forms.py:265:        self.helper = FormHelper()
dpnk/forms.py:434:        self.helper = FormHelper()
dpnk/forms.py:472:        self.helper = FormHelper()
dpnk/forms.py:554:        self.helper = FormHelper()
dpnk/forms.py:706:        self.helper = FormHelper()
dpnk/forms.py:1103:        self.helper = FormHelper()
t_shirt_delivery/admin_forms.py:20:from crispy_forms.helper import FormHelper
t_shirt_delivery/admin_forms.py:33:        self.helper = FormHelper()
````

**Comment by timthelion** [comment]
I'm really not sure how to work on debugging this or creating a minimum reproducible example as the backtrace doesn't even touch are code at all. 1.7.2 works fine for us. I think that it would be best at this point to wait and see if anyone else comes across this and has a simpler case.

**Comment by carltongibson** [comment]
OK.... — Ideally you'd stick a breakpoint in at the error point and let us know the value of template_name at least... 🙂 Short of more info, I'm not sure what to say.

**Comment by carltongibson** [comment]
5bfc957b966aed60d6650d622dde617a717293e8 uses the `default` filter. https://docs.djangoproject.com/en/2.2/ref/templates/builtins/#std:templatefilter-default

`field_template` defaults to none...

https://github.com/django-crispy-forms/django-crispy-forms/blob/649a407eb20d04ce345dc0bbfebf205589cc6f47/crispy_forms/helper.py#L216

so the default, e.g. `"bootstrap/field.html"` should be used. 

What's happening in your case?

**Comment by smithdc1** [comment]
@timthelion did you make any progress with this issue?

**Comment by timthelion** [comment]
@smithdc1 
Nope:
https://github.com/auto-mat/do-prace-na-kole/blob/devel/Pipfile#L32

**Comment by smithdc1** [comment]
@timthelion 

Have you given 1.8.1 a go? The below pr has been merged since 1.8.0. 🤞

https://github.com/django-crispy-forms/django-crispy-forms/pull/926
